hpx/runtime/serialization/detail/polymorphic_intrusive_factory.hpp

```diff
@@ -0,0 +1,213 @@
+//  Copyright (c) 2014 Anton Bikineev
+//  Copyright (c) 2014 Thomas Heller
+//
+//  Distributed under the Boost Software License, Version 1.0.
+//  See accompanying file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef HPX_SERIALIZATION_POLYMORPHIC_INTRUSIVE_FACTORY_HPP
+#define HPX_SERIALIZATION_POLYMORPHIC_INTRUSIVE_FACTORY_HPP
+
+#include <hpx/runtime/serialization/serialization_fwd.hpp>
+
+#include <hpx/util/jenkins_hash.hpp>
+#include <hpx/util/safe_lexical_cast.hpp>
+#include <hpx/util/static.hpp>
+
+#include <boost/preprocessor/stringize.hpp>
+#include <boost/noncopyable.hpp>
+#include <boost/unordered_map.hpp>
+#include <boost/atomic/atomic.hpp>
```

Comment:
Should be just `<boost/atomic.hpp>` for backward compatibility. See it fail with older boost versions here http://hermione.cct.lsu.edu/builders/hpx_gcc_4_6_boost_1_49_debian_x86_64_debug/builds/266/steps/build_core/logs/stdio